js-rattler/crate/solve.rs

```diff
@@ -61,20 +81,107 @@ pub async fn simple_solve(
             ..rattler_repodata_gateway::ChannelConfig::default()
         })
         .finish();
+
     let repodata = gateway
         .query(channels, platforms, specs.iter().cloned())
         .recursive(true)
         .execute()
         .await?;
 
-    // Solve
+    // We need this to find depends for locked packages
+    let repodata_keys: HashMap<(String, String, String), Vec<String>> = repodata
+        .iter()
+        .flat_map(|r| r.iter())
+        .map(|rec| {
+            let name = rec.package_record.name.as_normalized().to_string();
+            let version = rec.package_record.version.to_string();
+            let build = rec.package_record.build.clone();
+            ((name, version, build), rec.package_record.depends.clone())
+        })
+        .collect();
+
+    let installed_packages: Option<Vec<RepoDataRecord>> =
+        if locked_packages.is_null() || locked_packages.is_undefined() {
+            None
+        } else {
+            let js_locked_packages: Vec<JsLockedPackage> =
+                serde_wasm_bindgen::from_value(locked_packages).map_err(JsError::from)?;
+
+            let records = js_locked_packages
+                .into_iter()
+                .map(|pkg| {
+                    let url = Url::parse(&pkg.url)
+                        .map_err(|e| JsError::from(ParseChannelError::from(e)))?;
+
+                    let rec = PackageRecord {
+                        name: PackageName::try_from(pkg.package_name.clone())?,
+                        version: Version::from_str(&pkg.version)?.into(),
+                        build: pkg.build.clone(),
+                        build_number: pkg.build_number,
+                        subdir: "unknown".to_string(),
```

Comment:
Packages can also come from different subdirectories (e.g. `noarch` vs `emscripten-wasm`). The unique key of a package is `(name, version, build, subdir)`. You also do not have the `subdir` available to you from the input. This might cause problems down the road but Id be fine merging as is.